docs/how.rst

```diff
@@ -0,0 +1,256 @@
+
+
+.. _plugins:
+
+
+Plugins
+=======
+
+The Certbot client supports two types of plugins for
+obtaining and installing certificates: authenticators and installers.
+
+Authenticators are plugins used with the ``certonly`` command to obtain a certificate.
+The authenticator satisfies a challenge to validate that you
+control the domain(s) you are requesting a certificate for, obtains a certificate for the specified
+domain(s), and places the certificate in the ``/etc/letsencrypt`` directory on your
+machine. The authenticator does not install the certificate (it does not edit any of your server's configuration files to serve the
+obtained certificate). If you specify multiple domains to authenticate, they will
+all be listed in a single certificate. To obtain multiple separate certificates
+you will need to run Certbot multiple times.
+
+
+Installers are Plugins used with the ``install`` command to install a certificate.
+These plugins can modify your webserver's configuration to
+serve your website over HTTPS using certificates obtained by certbot.
+
+Plugins that do both can be used with the ``certbot run`` command, which is the default
+when no command is specified. The ``run`` subcommand can also be used to specify
+a combination_ of distinct authenticator and installer plugins.
+
+=========== ==== ==== =============================================================== =============================
+Plugin      Auth Inst Notes                                                           Challenge types (and port)
+=========== ==== ==== =============================================================== ===========================
+apache_     Y    Y    | Automates obtaining and installing a certificate with Apache  tls-sni-01_ (443)
+                      | 2.4 on Debian-based distributions with ``libaugeas0`` 1.0+.
+webroot_    Y    N    | Obtains a certificate by writing to the webroot directory of  http-01_ (80)
+                      | an already running webserver.
+nginx_      Y    Y    | Automates obtaining and installing a certificate with Nginx.  tls-sni-01_ (443)
+                      | Shipped with Certbot 0.9.0.
+standalone_ Y    N    | Uses a "standalone" webserver to obtain a certificate.        http-01_ (80) or
+                      | Requires port 80 or 443 to be available. This is useful on    tls-sni-01_ (443)
+                      | systems with no webserver, or when direct integration with
+                      | the local webserver is not supported or not desired.
+manual_     Y    N    | Helps you obtain a certificate by giving you instructions to  http-01_ (80),
+                      | perform domain validation yourself. Additionally allows you   dns-01_ (53) or
+                      | to specify scripts to automate the validation task in a       tls-sni-01_ (443)
+                      | customized way.
+=========== ==== ==== =============================================================== =============================
```

Comment:
`how.rst:44: ERROR: Malformed table.
Bottom/header table border does not match top border.`